src/node_kernels/VOFMassBDFNodeKernel.C

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+
+#include "node_kernels/VOFMassBDFNodeKernel.h"
+#include "Realm.h"
+
+#include "stk_mesh/base/MetaData.hpp"
+#include "stk_mesh/base/Types.hpp"
+#include "utils/StkHelpers.h"
+#include "utils/FieldHelpers.h"
+
+namespace sierra {
+namespace nalu {
+
+VOFMassBDFNodeKernel::VOFMassBDFNodeKernel(
+  const stk::mesh::BulkData& bulk, ScalarFieldType* scalarQ)
+  : NGPNodeKernel<VOFMassBDFNodeKernel>()
+{
+  const auto& meta = bulk.mesh_meta_data();
+
+  scalarQNID_ =
+    scalarQ->field_of_state(stk::mesh::StateN).mesh_meta_data_ordinal();
+
+  if (scalarQ->number_of_states() == 2)
+    scalarQNm1ID_ = scalarQNID_;
+  else
+    scalarQNm1ID_ =
+      scalarQ->field_of_state(stk::mesh::StateNM1).mesh_meta_data_ordinal();
+
+  scalarQNp1ID_ =
```

Comment:
Not for this PR, but I really want to rethink this state reassignment concept.  It functions well, and is clever.  I'm just not sure how I feel about changing the states like this instead of just having the states in the first place and then populating and updating them. I think we are saving essentially 2 fields since it is really only impacting the `dual_nodal_volume`.  Thoughts @rcknaus ?

Reply:
I think I'd just prefer having three states always and just multiply by zero in the case of bdf2.